app/src/main/java/fi/project/petcare/ui/screens/PetProfileFormScreen.kt

```diff
@@ -1,20 +1,252 @@
+            },
+            label = { Text("Weight") },
+            leadingIcon = { Icon(imageVector = Icons.Default.MonitorWeight, contentDescription = null) },
+            isError = weightError.isNotEmpty(),
+            singleLine = true,
+            modifier = Modifier.padding(bottom = 8.dp)
+        )
+        Text(
+            text = weightError,
+            color = Color.Red,
+            fontSize = 12.sp,
+            modifier = Modifier.padding(start = 16.dp)
+        )
+
+        // Species field with validation
+        OutlinedTextField(
+            value = species,
+            onValueChange = {
+                species = it
+                speciesError = if (it.isEmpty()) "Species cannot be empty" else ""
+            },
+            label = { Text("Species") },
+            leadingIcon = { Icon(imageVector = Icons.Default.Animation, contentDescription = null) },
+            isError = speciesError.isNotEmpty(),
+            singleLine = true,
+            modifier = Modifier.padding(bottom = 8.dp)
+        )
+        Text(
+            text = speciesError,
+            color = Color.Red,
+            fontSize = 12.sp,
+            modifier = Modifier.padding(start = 16.dp)
+        )
+
+        // Breed field with validation
+        OutlinedTextField(
+            value = breed,
+            onValueChange = {
+                breed = it
+                breedError = if (it.isEmpty()) "Breed cannot be empty" else ""
+            },
+            label = { Text("Breed") },
+            leadingIcon = { Icon(imageVector = Icons.Default.Pets, contentDescription = null) },
+            isError = breedError.isNotEmpty(),
+            singleLine = true,
+            modifier = Modifier.padding(bottom = 8.dp)
+        )
+        Text(
+            text = breedError,
+            color = Color.Red,
+            fontSize = 12.sp,
+            modifier = Modifier.padding(start = 16.dp)
+        )
+
+        // Birthdate field with validation
+        OutlinedTextField(
+            value = age_months,
+            onValueChange = {
+                age_months = it
+                age_monthsError = if (it.isEmpty()) "Age cannot be empty" else ""
+            },
+            label = { Text("Age") },
+            leadingIcon = { Icon(imageVector = Icons.Default.CalendarViewMonth, contentDescription = null) },
+            isError = age_monthsError.isNotEmpty(),
+            singleLine = true,
+            modifier = Modifier.padding(bottom = 8.dp)
+        )
+        Text(
+            text = age_monthsError,
+            color = Color.Red,
+            fontSize = 12.sp,
+            modifier = Modifier.padding(start = 16.dp)
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        // Save button
+        Button(
+            onClick = { /* Save the form data if validation passes */ },
+            modifier = Modifier
+                .fillMaxWidth()
+                .heightIn(48.dp),
```

Comment:
Consider using a height of at least 58.dp, as it fits better with all the buttons displayed in other screens.